client/packages/cli/commands/register/substrate.ts

```diff
@@ -32,6 +33,8 @@ const registerRelaychain = async (circuit: ApiPromise, target: ApiPromise, gatew
         gateway_genesis: await createGatewayGenesis(circuit, target),
         gateway_sys_props: createGatewaySysProps(circuit, gatewayData.registrationData.gatewaySysProps),
         allowed_side_effects: circuit.createType('Vec<AllowedSideEffect>', gatewayData.registrationData.allowedSideEffects),
+        parent_gateway_id: circuit.createType('Option<ChainId>', gatewayData.registrationData.parentChainId),
+        execution_layer: circuit.createType('ExecutionLayer', gatewayData.registrationData.executionLayer),
```

Comment:
Prefer if u take ExecutionLayer from PR#667 - t3rn_abi::codec (either Rlp / Scale for now)

Reply:
Are we sure those will always stay this way? I need something like this in executor to map gateways to correct tx/proof generation logic